### PR TITLE
Test getActiveFederation method from FederationSupportImpl

### PR DESCRIPTION
**Pull request by julia-zack**
Refactor existing getActiveFederation tests to use InMemoryStorage and FederationSupportBuilder
Add a test case to check for null new fed but existing old fed

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 1]
```diff
@@ -195,7 +204,7 @@ void getActiveFederatorPublicKeyOfType() {
         assertThrows(IndexOutOfBoundsException.class, () -> federationSupport.getActiveFederatorPublicKeyOfType(-1, KeyType.BTC));
     }
 
-    private Federation getNewFakeFederation(long creationBlockNumber) {
+    private Federation createFederation(long creationBlockNumber) {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:207]: Can we remove this and make use of FederationTestUtils in case? I think it's better we have utility methods centralized and we avoid re-creating them many times.

Also, as much as possible use an ERP Federation for testing. It's more realistic since it's what we have now

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 2]
```diff
@@ -79,10 +90,9 @@ void whenNewFederationIsNullThenActiveFederationIsGenesisFederation() {
     }
 
     @Test
-    void whenOldFederationIsNullThenActiveFederationIsNewFederation() {
-        Federation newFederation = getNewFakeFederation(100);
-        when(storageProvider.getNewFederation(federationMainnetConstants, activations)).thenReturn(newFederation);
-        when(storageProvider.getOldFederation(federationMainnetConstants, activations)).thenReturn(null);
+    void getActiveFederation_withNewFederationAndNullOldFederation_returnsNewFederation() {
+        Federation newFederation = createFederation(100);
+        storageProvider.setNewFederation(newFederation);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:95]: And the old federation? According to the test name there should be one

Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:95]: Guess we are also missing the test where there is only the new federation and not the old one

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 3]
```diff
@@ -124,23 +134,22 @@ void whenOldAndNewFederationArePresentReturnActiveFederationByActivationAge(
...
-        );
+        FederationSupport federationSupport = federationSupportBuilder
+            .withFederationConstants(federationConstants)
+            .withFederationStorageProvider(storageProvider)
+            .withRskExecutionBlock(executionBlock)
+            .withActivations(activations)
+            .build();
+
+        Federation newFederation = createFederation(newFedCreationBlockNumber);
+        Federation oldFederation = createFederation(oldFedCreationBlockNumber);
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:150]: This pretty much creates the same federation for both, so probably not very good idea for the purpose of the test

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:95]: that is this test, with a "real" new federation and a null old one

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 3 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:150]: yep, very good point

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 1 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java]: done

**Review by apancorb — APPROVED** [review]

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java [hunk 4]
```diff
@@ -50,27 +54,36 @@ class FederationSupportImplTest {
...
+            .withFederationStorageProvider(storageProvider)
+            .build();
+    }
+
+    @Test
+    void getActiveFederation_withNullFederations_returnsGenesisFederation() {
+        Federation genesisFederation = FederationTestUtils.getGenesisFederation(federationMainnetConstants);
+        Federation activeFederation = federationSupport.getActiveFederation();
+
+        assertThat(activeFederation, is(genesisFederation));
```
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:80]: i agree

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:97]: Looks like I got mixed up, you are right

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java [hunk 5]
```diff
@@ -69,6 +69,30 @@ public static ErpFederation getErpFederation(NetworkParameters networkParameters
...
+        final List<BtcECKey> erpSigners = BitcoinTestUtils.getBtcEcKeysFromSeeds(
+            new String[]{"fb01", "fb02", "fb03"}, true
+        );
+
+        List<FederationMember> fedMember = FederationTestUtils.getFederationMembersWithBtcKeys(
+            fedSigners);
+
+        FederationArgs federationArgs = new FederationArgs(
+            fedMember,
+            ZonedDateTime.parse("1970-01-18T12:49:08.400Z").toInstant(),
```
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java:82]: thanks :)

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java [hunk 6]
```diff
@@ -69,6 +69,30 @@ public static ErpFederation getErpFederation(NetworkParameters networkParameters
         );
     }
 
+    public static ErpFederation getErpFederation(NetworkParameters networkParameters, List<BtcECKey> fedSigners, long creationBlockNumber) {
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java:72]: Perhaps we could consider creating a FederationBuilder, what do you think? For another task obviously

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java:82]: thanks! :)

**Review by marcos-iov — APPROVED** [review]

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java (hunk 6 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationTestUtils.java:72]: yes, very good idea. I'll create the task. Thanks :)

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java (hunk 2 above)
Comment (julia-zack) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java:97]: no worries!
